privacyidea/lib/smsprovider/HttpSMSProvider.py

```diff
@@ -46,13 +48,26 @@
 from privacyidea.lib import _
 import requests
 from urlparse import urlparse
-
+import re
 import logging
 log = logging.getLogger(__name__)
 
 
 class HttpSMSProvider(ISMSProvider):
 
+    def _mangle_phone(self, phone, config):
+        regexp = config.get("REGEXP")
+        if regexp:
+            try:
+                m = re.match("^/(.*)/(.*)/$", regexp)
```

Comment:
Hmm, since we use `/` as the separator between match and replacement and we use greedy matching using `(.*)` here, we have the effect that having `/` in the match part works (like in the example), but having `/` in the replacement part will break. But I doubt anyone would want to use `/` in the replacement expression?

Reply:
Are you sure? 
Very complex regexp will not work anyway. You will not be able to use something like this in perlre

    /([0-9]+).*/\1/

Reply:
@zerealfox thats right. The "/" will work to _find_ it, but not to _replace_ it.

+32 123 / 345 -> 32123345 will work

+32 123 345 -> 32123/345 will not work

Reply:
Hi Pascal, this use case makes a lot of sense and this PR implements it properly. Above, I was wondering about something else: Might there be a case in which some pattern is used as the first part of regexp, and it should be replaced with "/"? An example would be "replace spaces in the phone number with slashes". This will not work as expected, as we use ``/`` to separate match and replacement.